packages/cli/src/commands/extensions/configure.ts

```diff
@@ -0,0 +1,148 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import type { CommandModule } from 'yargs';
+import { ExtensionManager } from '../../config/extension-manager.js';
+import {
+  getEnvContents,
+  updateSetting,
+  promptForSetting,
+} from '../../config/extensions/extensionSettings.js';
+import { loadSettings } from '../../config/settings.js';
+import { requestConsentNonInteractive } from '../../config/extensions/consent.js';
+
+// --- SET COMMAND ---
+interface SetArgs {
+  name: string;
+  setting: string;
+}
+
+const setCommand: CommandModule<object, SetArgs> = {
+  command: 'set <name> <setting>',
+  describe: 'Set a specific setting for an extension.',
+  builder: (yargs) =>
+    yargs
+      .positional('name', {
+        describe: 'Name of the extension to configure.',
+        type: 'string',
+        demandOption: true,
+      })
+      .positional('setting', {
+        describe: 'The setting to configure (name or env var).',
+        type: 'string',
+        demandOption: true,
+      }),
+  handler: async (args) => {
+    const { name, setting } = args;
+    const { extension, extensionManager } = await getExtensionAndManager(name);
+    if (!extension || !extensionManager) {
+      return;
+    }
+    const extensionConfig = extensionManager.loadExtensionConfig(
+      extension.path,
+    );
+    if (!extensionConfig) {
+      console.error(`Could not find configuration for extension "${name}".`);
+      return;
+    }
+    await updateSetting(
+      extensionConfig,
+      extension.id,
+      setting,
+      promptForSetting,
+    );
+  },
+};
+
+async function getExtensionAndManager(name: string) {
```

Comment:
I would move this to a shared utils file under command/extensions and use it across the commands, it probably is basically identical code in all of them?